addons/binding/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/api/EtherRainCommunication.java

```diff
@@ -0,0 +1,257 @@
+/**
+ * Copyright (c) 2010-2019 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.etherrain.internal.api;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.HttpURLConnection;
+import java.net.InetAddress;
+import java.net.SocketTimeoutException;
+import java.net.URL;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.openhab.binding.etherrain.internal.api.EtherRainStatusResponse.CommandResult;
+import org.openhab.binding.etherrain.internal.api.EtherRainStatusResponse.CommandStatus;
+import org.openhab.binding.etherrain.internal.api.EtherRainStatusResponse.OperatingStatus;
+
+public class EtherRainCommunication {
+
+    private final static String BROADCAST_DISCOVERY_MESSAGE = "eviro_id_request1";
+    private final static int BROADCAST_DISCOVERY_PORT = 8089;
+
+    private final static String ETHERRAIN_USERNAME = "admin";
+
+    private final static int HTTP_TIMEOUT = 1000;
+
+    private final static int BROADCAST_TIMEOUT = 80;
+
+    private String address = null;
+    private int port = 80;
+    private String password = "pw";
+
+    public EtherRainCommunication() {
```

Comment:
do you need this constructor? it seems a bit strange that `address=null` should be any useful. if unsued, remove it. you can then remove the field initializations, too.